### ISSUE TITLE
enabled-proposed-api flag broken on insiders

### ISSUE THREAD

**Issue by daytonellwanger** [microsoft/vscode#81917]
If I try to run VS Code Insiders with the `--enable-proposed-api` flag I see this:
![image](https://user-images.githubusercontent.com/6383490/66156439-82b25c80-e5d6-11e9-9d00-f3196873cb8c.png)

```
bootstrap-window.js:212 [uncaught exception]: TypeError: (e.args.enable-proposed-api || []).map is not a function
onUnexpectedError @ bootstrap-window.js:212
bootstrap-window.js:215 TypeError: (e.args.enable-proposed-api || []).map is not a function
    at new x (abstractExtensionService.ts:460)
    at new e (abstractExtensionService.ts:86)
    at new B (extensionService.ts:75)
    at h._createInstance (instantiationService.ts:117)
    at h._createServiceInstance (instantiationService.ts:216)
    at h._createServiceInstanceWithOwner (instantiationService.ts:205)
    at h._createAndCacheServiceInstance (instantiationService.ts:194)
    at h._getOrCreateServiceInstance (instantiationService.ts:142)
    at h._createInstance (instantiationService.ts:94)
    at h.createInstance (instantiationService.ts:82)
    at t.WorkspaceService.acquireInstantiationService (configurationService.ts:305)
    at function.n.acquireInstantiationService.setTimeout (workbench.ts:207)
```
Is this flag deprecated?

**Comment by jrieken** [comment]
I cannot reproduce. What's the full command line that you are using?

/cc @aeschli who ran some refactorings in this area

**Comment by daytonellwanger** [comment]
I'm debugging a VS Code extension with this launch configuration:
```
{
    "version": "0.1.0",
    "configurations": [
        {
            "name": "Launch Extension",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": ["--extensionDevelopmentPath=${workspaceRoot}", "--enable-proposed-api", "extension.id" ],
            "stopOnEntry": false,
            "sourceMaps": true,
            "outFiles": [ "${workspaceRoot}/out/prod/**/*.js" ],
            "preLaunchTask": "compile-typescript",
            "smartStep": true,
            "pathMapping": { "/": "/" }
        }
    ]
}
```

**Comment by daytonellwanger** [comment]
Changing the configuration to
```
{
    "version": "0.1.0",
    "configurations": [
        {
            "name": "Launch Extension",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": ["--extensionDevelopmentPath=${workspaceRoot}"],
            "stopOnEntry": false,
            "sourceMaps": true,
            "outFiles": [ "${workspaceRoot}/out/prod/**/*.js" ],
            "preLaunchTask": "compile-typescript",
            "smartStep": true,
            "pathMapping": { "/": "/" }
        }
    ]
}
```
resolves the problem

**Comment by jrieken** [comment]
adding @isidorn to check if something has changed here

**Comment by jrieken** [comment]
Ok, with the provided launch config, I can reproduce. Unsure when that was broken...

**Comment by isidorn** [comment]
The only change in that area was done by @aeschli in the way how we parse `args`.
I can also reproduce this in stable, thus this is not a candidate (I just get a different error message).

Here's the unminified stack when running out of sources

```
/Users/isidor/Development/vscode/out/bootstrap-window.js:215 TypeError: (environmentService.args.enable-proposed-api || []).map is not a function
    at new ProposedApiController (abstractExtensionService.js:350)
    at new AbstractExtensionService (abstractExtensionService.js:48)
    at new ExtensionService (extensionService.js:25)
    at InstantiationService._createInstance (instantiationService.js:88)
    at InstantiationService._createServiceInstance (instantiationService.js:177)
    at InstantiationService._createServiceInstanceWithOwner (instantiationService.js:165)
    at InstantiationService._createAndCacheServiceInstance (instantiationService.js:156)
    at InstantiationService._getOrCreateServiceInstance (instantiationService.js:113)
    at InstantiationService._createInstance (instantiationService.js:69)
    at InstantiationService.createInstance (instantiationService.js:59)
    at WorkspaceService.acquireInstantiationService (configurationService.js:189)
    at setTimeout (workbench.js:115)
```

[Code pointer](https://github.com/Microsoft/vscode/blob/56c472e18511f483d80cb8c1b8bd7b35c467a1af/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L460) so the enable-proposed-api is no longer a string array.

@aeschli let me know if you do not know where the breakage might be and I can investigate further

fyi @weinand

**Comment by jrieken** [comment]
<img width="438" alt="Screenshot 2019-10-16 at 13 46 02" src="https://user-images.githubusercontent.com/1794099/66916344-50afea00-f01b-11e9-99ec-e1b55665ceee.png">

Looks like `enable-proposed-api` is set to `true` in that case.

**Comment by jrieken** [comment]
Leaving this to @aeschli  and @isidorn. Running with `--enable-proposed-api` (incomplete, no extension) from the command line works (e.g ignore the incomplete option) but somehow launching from the debugger circumvents this and assigns the boolean value. 

Re-assign back to me if I should manually check the value of my flag. Which I believe defeats the purpose of spelling out the expected types when defining the option

**Comment by weinand** [comment]
I'll investigate.

**Comment by weinand** [comment]
Currently extension debugging has the limitation that command line arguments with options must be passed in the form `--option=arg` in the launch config.

So instead of passing
```
"--enable-proposed-api", "extension.id"
```
it is necessary to pass this:
```
"--enable-proposed-api=extension.id"
```

**Comment by jrieken** [comment]
But it also seems to by pass our validation

**Comment by weinand** [comment]
Fixed via debt issue #80461
